examples/stable_diffusion_slackbot/README.md

```diff
@@ -0,0 +1,469 @@
+
+```python
+pip install oxenai
+```
+
+Our app listens for two main types of events: `file_created` and `reaction_added`. We can parse the event type we’re receiving from Slack and create a branching code path for each: 
+
+```python
+# in @app.route("/", methods=["POST]) from earlier 
+def post(): 
+	data = request.get_json()
+	if data["event"]["type"] == "file_created":
+		# TODO: Parse for file creation 
+		 
+	if data["event"]["type"] == "reaction_added":
+		# TODO: Parse for reactions 
+
+	# No other action types should be allowed, but catch-all in case
+	return "Success"
+```
+
+### Parsing file uploads and pushing to oxen
+
+**1. Validate event relevance**
+
+We want to ensure that we’re only parsing image uploads from our bot user, and not from any human (or other bot) users in the channels where our app lives.
+
+To do so, get the Slack `user_id` for our bot user by printing it (once) to the terminal, and store it locally as a `SLACK_BOT_USER_ID` environment variable. 
+
+Then, we can write a validation function that will let us know if we’re clear to proceed with parsing.
+
+```python
+def is_valid_file_upload(file_data: str) -> bool:
+	if file_data['event']['user_id'] != os.environ["SLACK_BOT_USER_ID"]:
+        print("File upload not by bot user, aborting")
+        return False
+   return True
+```
+
+**2. Fetch and download the image file from the Slack API**
+
+The event data Slack sends includes the ID of the file, but not the file object itself. We can use Slack’s python client to retrieve it given the `conversation_id` (which channel to search in) and `message_ts` (timestamp of the message, which Slack uses as a unique ID)
+
+```python
+def fetch_message(conversation_id, message_ts):
+	try:
+      result = client.conversations_history(
+          channel=conversation_id,
+          inclusive=True,
+          oldest = message_ts,
+          limit=1
+      )
+  except SlackApiError as e:
+      print(f"Error: {e}")
+  return result
+```
+
+This will return a `file` object with a `url_private_download` we can use to download the image file.
+
+```python
+def fetch_file(file_id):
+    try:
+        file = client.files_info(
+            file=file_id,
+            count=1
+        )
+    except SlackApiError as e:
+        print(f"Error: {e}")
+    return file
+
+def download_image(image_url, image_id):
+    img_data = requests.get(image_url, headers=headers).content
+    file_hash = hash_file_id(image_id)
+    file_path = f'{IMAGE_DIR}/{file_hash}.png'
+
+    with open(file_path, 'wb') as handler:
+        handler.write(img_data)
+    return file_path
+```
+
+**3. Upload to Oxen**
+
+At the top level of the server file, set up an oxen `RemoteRepo` to point at your target repository: 
+
+```python
+from oxen import RemoteRepo
+repo = RemoteRepo("your-namespace/your-repo")
+repo.checkout("your-branch", create=False) # or True for new branch
+```
+
+Then remotely commit the image: 
+
+```python
```

Comment:
This indentation looks funny, and don't we only want to commit if the image is successfully added?